### DISCUSSION TITLE
How to Build VSIX Package for vscode-cpptools Extension Across Multiple Platforms

### DISCUSSION THREAD

**Discussion by karthick-govindaraj** [microsoft/vscode-cpptools#12688] [General]
Hi All,
I’d like to know how to build the .vsix package for the vscode-cpptools extension from the source, targeting multiple platforms like Linux (x86), ARM, etc. What are the steps, dependencies, or changes needed for cross-compilation to support different architectures? Any guidance or resources would be greatly appreciated.

Thank you!

**Comment by bobbrow** [comment]
May we ask why you'd like to do this?  If you're planning to redistribute the language server binaries as your own custom offering, please note that our [runtime license](https://github.com/microsoft/vscode-cpptools/blob/main/RuntimeLicenses/cpptools-LICENSE.txt) prohibits that.  Only the TypeScript code in this repo is open source.

You are granted the rights to run and debug the extension on your personal computer if you intend to contribute to this repo ([CONTRIBUTING.md](https://github.com/microsoft/vscode-cpptools/blob/main/CONTRIBUTING.md))

**Reply by karthick-govindaraj** [reply]
Hi @bobbrow,

Sorry, I hadn't read the license properly. I'm running VS Code on a RISC-V architecture, so I was attempting to cross-compile the C/C++ extension source to run it on RISC-V systems in VS Code. Could you consider making a future release of the C/C++ extension for the RISC-V architecture? That would be extremely helpful for me as well.

**Reply by bobbrow** [reply]
Did you build your own version of VS Code for RISC-V?  I wasn't aware that such a version existed.  RISC-V isn't currently on our radar and we wouldn't begin building or distributing an extension that supports it until the VS Code team officially supported it.

**Reply by karthick-govindaraj** [reply]
Hi @bobbrow 

I am using Code Server, which, as you know, turns VSCode into a cloud-based IDE. The build I am using was generated unofficial sources: [GitHub issue](https://github.com/coder/code-server/issues/6951) and [GitLab release](https://gitlab.b-data.ch/coder/code-server/-/releases).

I am exploring the packages and trying to cross-compile extensions that are useful for us, which is why I'm attempting to cross-compile the [C++ extension](https://github.com/microsoft/vscode-cpptools) for RISC-V.

Is there any possibility of cross-compiling the C/C++ extension for RISC-V? That would be incredibly helpful, as it's such a great extension for developing C/C++ programs and projects.

**Reply by bobbrow** [reply]
I'm sorry to let you know that our license also prohibits using our extension in alternate distributions of VS Code - only the official one produced by Microsoft may be used. So even if we did produce a RISC-V version, you wouldn't be licensed to use it with Code Server.
